src/main/webapp/election-info.js

```diff
@@ -171,50 +169,67 @@ function initializeMap() {
-        center: results[0].geometry.location,
-        zoom: 13
-      });
-      console.log("Created map");
-    } else {
-      alert('Geocode was not successful for the following reason: ' + status);
-    }
-  });
-
   let servletUrl = "/polling-stations?electionId=" + urlParams.get("electionId");
 
   fetch(servletUrl)
     .then(response => response.json())
     .then((pollingStationList) => {
-      pollingStationList.forEach((pollingStation) => {
-        var type;
-        if (pollingStation.locationType == "dropOffLocation") {
-          type = "Drop off only.";
-        } else if (pollingStation.locationType == "earlyVoteSite") {
-          type = "Early vote site only.";
-        } else if (pollingStation.locationType == "pollingLocation") {
-          type = "Standard polling location.";
-        }
+      if (pollingStationList === undefined || pollingStationList.length == 0) {
+        document.getElementById("polling-stations-map").style.display = "none";
+        document.getElementById("no-polling-stations").innerText = 
+          "Sorry, we can't find any polling stations for this election near your address.";
+        console.log("displayed polling station message");
+        return;
+      }
 
-        const description =
-          '<div id="content">' +
-          '<div id="siteNotice">' +
-          "</div>" +
-          '<h2 id="firstHeading" class="firstHeading">'+ pollingStation.name + '</h2>' +
-          '<div id="bodyContent">' +
-          "<p>" + pollingStation.address + "</p>" +
-          "<p>Hours: Open " + pollingStation.pollingHours + " beginning " + pollingStation.startDate + 
-          " and until " + pollingStation.endDate + ".</p>" +
-          "<p>" + type + "</p>" +
-          "</div>" +
-          "</div>";
-
-        addPollingStation(pollingStation.address, map, pollingStation.name, description);
-        console.log("Added polling station marker");
+      document.getElementById("no-polling-stations").innerText = "";
+      document.getElementById("polling-stations-map").style.display = "block";
+      
+      const address = urlParams.get('address');
+      
+      geocoder.geocode( { 'address': address}, function(results, status) {
+        if (status == 'OK') {
+          map = new google.maps.Map(document.getElementById("polling-stations-map"), {
+            center: results[0].geometry.location,
+            zoom: 13
+          });
+          console.log("Created map");
+          
+          const homeMarker = new google.maps.Marker({
+            position:results[0].geometry.location, 
+            map:map,
+            icon: "https://img.icons8.com/ios-filled/50/000000/home.png",
+            draggable: false
+          });
+          console.log("Created user address marker");
+          
+          pollingStationList.forEach((pollingStation) => {
+
+            let descriptionTemplate =  
+              '<div id="content">' +
```

Comment:
maybe you can use a template literal instead?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
will do!